src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLSurfaceData.m

```diff
@@ -28,20 +28,11 @@
 #import "sun_java2d_metal_MTLSurfaceData.h"
 
 #import "jni_util.h"
-#import "MTLRenderQueue.h"
 #import "MTLGraphicsConfig.h"
 #import "MTLSurfaceData.h"
-#import "ThreadUtilities.h"
 #include "jlong.h"
 
-/**
- * The following methods are implemented in the windowing system (i.e. GLX
- * and WGL) source files.
- */
-extern jlong MTLSD_GetNativeConfigInfo(BMTLSDOps *bmtlsdo);
-extern jboolean MTLSD_InitMTLWindow(JNIEnv *env, BMTLSDOps *bmtlsdo);
-extern void MTLSD_DestroyMTLSurface(JNIEnv *env, BMTLSDOps *bmtlsdo);
-
+jboolean MTLSD_InitMTLWindow(JNIEnv *env, BMTLSDOps *bmtlsdo);
```

Comment:
How the MTLSD_WINDOW is used? Do we use it for the layer-based rendering?

Reply:
It is used in MTLContex.m. It just indicates a surface that we render to. We blit from this surface to the CAMetalLayer lateron.
It looks like MTLSD_WINDOW can be replaced with simple MTLSD_TEXTURE. This investigation and cleanup has already been identified during lanai code review (Bug - JDK-8263463)